lib/shopify_app/managers/webhooks_manager.rb

```diff
@@ -41,21 +41,23 @@ def add_registrations
           ShopifyAPI::Webhooks::Registry.add_registration(
             topic: attributes[:topic],
             delivery_method: attributes[:delivery_method] || :http,
-            path: attributes[:address],
-            handler: webhook_job_klass(attributes[:topic]),
+            path: attributes[:path] || attributes[:address],
+            handler: webhook_job_klass(attributes[:path]),
             fields: attributes[:fields]
           )
         end
       end
 
       private
 
-      def webhook_job_klass(topic)
-        webhook_job_klass_name(topic).safe_constantize || raise(ShopifyApp::MissingWebhookJobError)
+      def webhook_job_klass(path)
+        webhook_job_klass_name(path).safe_constantize || raise(ShopifyApp::MissingWebhookJobError)
       end
 
-      def webhook_job_klass_name(topic)
-        [ShopifyApp.configuration.webhook_jobs_namespace, "#{topic.gsub("/", "_")}_job"].compact.join("/").classify
+      def webhook_job_klass_name(path)
+        job_file_name = path.split("/").last
```

Comment:
```suggestion
        job_file_name = Pathname(path.basename.to_s)
```
(Just a little higher level).